packages/docusaurus/src/server/__tests__/configValidation.test.ts

```diff
@@ -0,0 +1,164 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import * as Joi from '@hapi/joi';
+import {ConfigSchema, DEFAULT_CONFIG} from '../configValidation';
+
+describe('validateConfig', () => {
```

Comment:
probably worth to add a test with an empty config so that we snapshot the error for required fields.

BTW, I wonder if we shouldn't make some of these optional. Like `baseUrl: '/'` seems a decent default value...

Reply:
baseUrl can be optional, I think `url` can't be optional and making title optional should be fine.

Reply:
empty config will throw error for single filed, we need to use `abortEarly:true`  for all error.